packages/mp/src/mp/validate_project/pre_build_validation.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import dataclasses
+import pathlib
+from collections.abc import Callable
+
+import mp.core.file_utils
+import mp.core.unix
+from mp.common.exceptions import NonFatalExceptionError
+
+
+@dataclasses.dataclass(slots=True, frozen=True)
+class PreBuildValidations:
+    integration_path: pathlib.Path
+    logs: list[str] = dataclasses.field(default_factory=list)
+
+    def get_logs(self) -> list[str]:
+        """Return the log object that stores all the logs inside it.
+
+        Returns:
+            list[str]: A list containing all the logs.
+
+        """
+        return self.logs
+
+    def is_all_validation_passed(self) -> bool:
```

Comment:
This definitely can be a property 
```suggestion
        @property
        def all_validations_passed(self) -> bool:
```

Then you can call it without parenthesis, like an attribute.